mlir-tensorrt/compiler/lib/Dialect/TensorRTRuntime/IR/TensorRTRuntime.cpp

```diff
@@ -113,6 +113,118 @@ void EnqueueOp::getEffects(
   }
 }
 
+//===----------------------------------------------------------------------===//
+// EnqueueAllocOp
+//===----------------------------------------------------------------------===//
+
+void EnqueueAllocOp::inferOperandKind(
+    ArrayRef<TensorKindLattice *> operands,
+    ArrayRef<const TensorKindLattice *> results,
+    llvm::function_ref<void(OpOperand &, TensorKind)> setOperandKind) {
+  for (OpOperand &operand : getOperation()->getOpOperands()) {
+    if (!isa<ShapedType>(operand.get().getType()))
+      continue;
+    if (isOperandOnHost(&operand)) {
+      setOperandKind(operand, TensorKind::Host);
+      continue;
+    }
+    setOperandKind(operand, TensorKind::Device);
+  }
+}
+
+TensorKind EnqueueAllocOp::getStaticOperandTensorKind(OpOperand &operand) {
+  return isOperandOnHost(&operand) ? TensorKind::Host
+                                                     : TensorKind::Device;
+}
+
+static bool hasCanonicalStrides(MemRefType memrefType) {
+  if (auto layoutAttr =
```

Comment:
You can just use 

```
memrefType.getLayout().isIdentity()
```

Reply:
You can't rely on `StridedLayoutAttr` since the layout can take different forms (AffineMap or StridedLayoutAttr).